packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -131,7 +141,109 @@ def test_metrics_agents_statsd(dcos_api_session):
         with dcos_api_session.marathon.deploy_and_cleanup(marathon_app, check_health=False):
             endpoints = dcos_api_session.marathon.get_app_service_endpoints(marathon_app['id'])
             assert len(endpoints) == 1, 'The marathon app should have been deployed exactly once.'
-            get_metrics_prom(dcos_api_session, agent, expected_metrics)
+
+            @retrying.retry(wait_fixed=2000, stop_max_delay=300 * 1000)
+            def check_statsd_metrics(response):
+                response = get_metrics_prom(dcos_api_session, agent)
+                for metric_name in expected_metrics:
+                    assert metric_name in response.text
+            check_statsd_metrics()
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=300 * 1000)
+def get_task_hostname(dcos_api_session, framework_name, task_name):
+    mesos_id = node = ''
+    state_response = dcos_api_session.get('/state', host=dcos_api_session.masters[0], port=5050)
+    assert state_response.status_code == 200
+    state = state_response.json()
+
+    for framework in state['frameworks']:
```

Comment:
I think I've seen it before [here](https://github.com/dcos/dcos/pull/3764/files#diff-a99db445cccf205fe11239e4be6de2aaR157) can you extract it to function?

Reply:
sorry, all 4 of these dcos prs I have open are all basically adding the same code - let me try to consolidate them better and get as similar versions as I can in all of these